File: raeunlee/week_14/PGS_네트워크.py
from collections import deque

def solution(n,computers):
    answer = 0
    visited = [False for _ in range(n)]
    for i in range(n):
        if visited[i] == False:
            visited[i]=True
            q = deque([i])
            answer += 1
        while q:
            idx = q.popleft()
            visited[idx]=True
            for j in range(n):
                if visited[j] == False and computers[idx][j] == 1:
                    q.append(j)
    return answer

File: hejoo345/week_14/PGS_네트워크.java
class Solution {
    static int n;
    static int[][] computers;
    static boolean[] visited;

    public int solution(int cn, int[][] ccomputers) {
        n = cn;
        computers = ccomputers;
        visited = new boolean[n];
        int answer = 0;
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                answer++;
                dfs(i);
            }
        }
        return answer;
    }

    public void dfs(int node) {
        if (visited[node]) return;

        visited[node] = true;
        for (int i = 0; i < n; i++) {
            if (computers[node][i] == 1) {
                dfs(i);
            }
        }
    }
}